packages/datadog-plugin-next/src/index.js

```diff
@@ -62,6 +62,11 @@ class NextPlugin extends ServerPlugin {
 
     this.config.hooks.request(span, req, res)
 
+    // propagate up web resource name if changed in the request hooks
+    const fullResource = span.context()._tags['resource.name']
+    const resource = fullResource.split(' ')[1] // GET x/y/z --> x/y/z
+    web.setRoute(req, resource)
```

Comment:
I also wondered about the idea of having an option for collapsing resource names behind a config option (like `collapseErrorResources`, or something), but thought an approach like this one would be more straightforward. Is the problem with this approach that `resource.name` should not be changed at all by the user, for the reasons you identified, like hiding information? In that case, would any approach, like adding an option to collapse mentioned above, that modified `resource.name` from the actual resource name be breaking?

Reply:
Also, changing the resource name of a span shouldn't change the resource name of another span. The only reason this was done for http by merging it downwards is because http doesn't have a resource name to begin with. It seems like we might be working around a product requirement with a hack in the library?

Reply:
Ah OK, I think I understand now, that wasn't clear to me before. I think the problem here with Next.js is that, since it's wrapped inside of a server and isn't itself a server, users who want to change the resource name on a Next.js span in the hooks and also have it appear on the parent server span (`http`, `express`, etc.) would have to not only use the `next` span hooks, but the appropriate span hooks for the server it's wrapped in as well (I believe by default it's just an `http` server in the Next.js internals, but in a custom server definition could be whatever). 

If that's the case, then that can be the general recommendation (using `next` span hooks and the appropriate parent framework span hooks). Otherwise, if we wanted to provide a more direct approach through _only_ our Next.js integration, would adding a config option to collapse error resource names into `/404` or `/500` be OK, since it wouldn't be changing the name of another span after the fact? Or does that still cause issues with hiding information (from what I've seen, though, automatic collapsing on error resources still allows queryable URLs of the original resource in the Datadog UI, because of the `path_group`)

Reply:
At that point we might as well have an option to just collapse the spans completely. To be honest that might make sense, because for Next it's not always clear whether the server span is actually useful. If the user has an existing server that wraps Next with its own routes, then it makes sense to keep them separate. But a lot of users also let Next handle everything on its own and in that case having 2 spans ends up being unnecessary and somewhat confusing. Not sure of the complexity of implementing this though, especially with the current state of web.js

Reply:
I agree that maybe it's a good idea to try and change this behavior at some point when we can dedicate some resources to it. With that in mind, plus the worries of accidentally hiding information from spans with a change like this, the general solution for users that wish to do this kind of resource name collapsing (whatever path name into a `GET /404` or something) is to use the relevant server framework (custom server or `http` for default Next.js one) hooks. I'll close this PR out for now, and for cases like this will recommend using hooks as they are now with the caveat of hiding information. Otherwise, it will need to be a FR for the kind of default behavior you described.